### PR TITLE
Add a test for the login flow when there is a teamserver

### PR DESCRIPTION
**Pull request by richvdh**
- just to check it keeps working.

Requires https://github.com/matrix-org/matrix-react-sdk/pull/1098

### CODE REVIEW

File: test/app-tests/loading.js [hunk 1]
```diff
@@ -343,6 +346,40 @@ describe('loading:', function () {
             }).done(done, done);
         });
 
+        it("registers correctly with a Riot Team Server", function() {
```
Comment (lukebarnard1) [inline test/app-tests/loading.js]: This should really be "logs in to the Riot Team Server". I would be awesome to test the slightly more important flow of a user registering with a team email address and being tracked as a referral!

File: test/app-tests/loading.js [hunk 2]
```diff
@@ -343,6 +346,40 @@ describe('loading:', function () {
...
+            sdk.setFetch(httpBackend.fetchFn); // XXX: ought to restore this!
```
Comment (lukebarnard1) [inline test/app-tests/loading.js]: Also, I haven't convinced myself how this is getting used but I shall look when I get in this morning!

Comment (lukebarnard1) [inline test/app-tests/loading.js]: So I'm pretty convinced that `sdk` should be `RtsClient`

Comment (richvdh) [inline test/app-tests/loading.js]: no... it goes through [here](https://github.com/matrix-org/matrix-react-sdk/pull/1098/files#diff-1fdf421c05c1140f6d71444ea2b27638R32).

(it turns out that if I `import RtsClient` directly, then I get a *different* RtsClient to that being used by the react-sdk, which means that replacing its fetchFn doesn't work. I think that's due to https://github.com/webpack/webpack/issues/2740. In the end I decided it probably makes sense to replace the fetchfn for the whole of the sdk anyway.)

Comment (lukebarnard1) [inline test/app-tests/loading.js]: Ah, in which case I'd suggest renaming the one exported in the sdk to be `setRtsFetchFn`. and it doesn't look like you're replacing fetchfn for the whole of the sdk ...

Comment (richvdh) [inline test/app-tests/loading.js]: well, I'm not, but it's only used in RtsClient and autocomplete/DuckDuckGoProvider.js for now. The thinking is that we can add more things to sdk.setFetch if and when it is useful.

Comment (lukebarnard1) [inline test/app-tests/loading.js]: ah ok, np

**Pull request merged**